### ISSUE TITLE
Incorrect transports already unloaded error message

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#874]
Extracted and summarizing from #553

Setting - six transports with six infantry, three tanks, three artillery
(i.e. three transports have inf/art, three have inf/tank) loaded in them, ready to unload. You see this problem almost every game in revised with UK unloading units into Europe.

3) click a infantry and an arty and move them to the land. 
3a) unload the inf/arty.
3b) [actual bug] now after jumping through all the hoops, I've selected the correct options to unload one infantry and one artillery from one transport into one territory. I now select one tank and try to move it into a different territory. But I get an error: "Transport has already unloaded units into [first territory]" How can this be? Even if the game disregarded my selections entirely and insisted on moving an infantry from one of the inf/tank combos, there should still be two tanks available to unload here.

**Comment by ron-murhammer** [comment]
@DanVanAtta I think this issue might be resolved now as well. Either that or you need a more complex use case to hit it now that default unloading is much better.

**Comment by DanVanAtta** [comment]
Let's report testing results here, good area to put some extra (manual) test coverage

**Comment by DanVanAtta** [comment]
Still broken, but in a different way:

- Load 6 infantry, 3 tanks, and 3 arty onto 6 transports
- Select one inf and one arty, and unload them
- Confirm the transport unload prompt, unloading from one transport
- Select one tank, try to unload it:
 - note that you cannot unload in any territory except the one just landed in
 - expected: should be able to move the tank anywhere, the infantry and arty were together on the one trasnport


Bug: So instead of moving the one infantry and one arty from the same trasnport, the game chose an infantry that was with a tank, and an arty that was with a different infantry.

Then, when clicking the 3 tanks, tripleA selects the top one in the stack, which is the only tank with the move restriction.

Bottom line, the transport unload prompt is broken, even though unloading from one transport, can still get two units from two different trasnports anyways.

**Comment by ron-murhammer** [comment]
This should be addressed with https://github.com/triplea-game/triplea/pull/4835

**Comment by panther2** [comment]
@ron-murhammer I hope to find some time for testing soon.

**Comment by panther2** [comment]
@ron-murhammer I have recreated @DanVanAtta 's scenario (hoping to have it understood correctly).
I tested using 1.10.14782 - and it looks like unloading the transports is resolved correctly, now.

**Comment by ron-murhammer** [comment]
Awesome. Closing this issue now! Only took 3 years!
